### PR TITLE
cam6_3_122: Aerosol optics updates

### LINKED ISSUE #865: Aerosol optics answer changes
### Issue Type

Other (please describe below)

### Issue Description

Isolate modifications in the generalization of aerosol optics (proposed in PR #824) that change results.  The changes in results are expected to be small -- near roundoff level.  This includes generalization of history fields to account for a large number of bins in sectional aerosol models.

### Will this change answers?

Yes

### Will you be implementing this yourself?

Yes

### PR DESCRIPTION
**Pull request by fvitt**
closes #865 

Answer changes are expected to be near roundoff

### CODE REVIEW

**Review by cacraigucar — APPROVED** [review]

**Review by nusbaume — CHANGES_REQUESTED** [review]
Looks good!  Just have a couple change requests and a question.

File: src/physics/cam/modal_aer_opt.F90 [hunk 1]
```diff
@@ -231,47 +237,54 @@ subroutine modal_aer_opt_init()
    call addfld ('EXTxASYMdn',   (/ 'lev' /), 'A','  ','extinction 550 * asymmetry factor, day night',        &
                 flag_xyfill=.true.)
 
+   call addfld ('AODTOT', horiz_only, 'A','1','Aerosol optical depth summed over all sw wavelenghts', &
+                flag_xyfill=.true.)
+   call addfld ('AODTOTdn', horiz_only, 'A','1','Aerosol optical depth summed over all sw wavelenghts', &
+                flag_xyfill=.true.)
```
[the remark this answers is not in the document]
Comment (fvitt) [reply src/physics/cam/modal_aer_opt.F90]: fixed

File: src/physics/cam/modal_aer_opt.F90 [hunk 2]
```diff
@@ -429,6 +435,11 @@ subroutine modal_aer_opt_init()
          call addfld ('AODABS'//diag(ilist),   horiz_only,  'A','  ', &
               'Aerosol absorption optical depth 550 nm', flag_xyfill=.true.)
 
+         call addfld ('AODTOT'//diag(ilist), horiz_only, 'A','1', &
+              'Aerosol optical depth summed over all sw wavelenghts', flag_xyfill=.true.)
+         call addfld ('AODTOTdn'//diag(ilist), horiz_only, 'A','1', &
+              'Aerosol optical depth summed over all sw wavelenghts', flag_xyfill=.true.)
```
[the remark this answers is not in the document]
Comment (fvitt) [reply src/physics/cam/modal_aer_opt.F90]: fixed

File: src/physics/cam/modal_aer_opt.F90 [hunk 3]
```diff
@@ -983,7 +1007,7 @@ subroutine modal_aero_sw(list_idx, state, pbuf, nnite, idxnite, &
                   ! write(iulog,*) 'itab,jtab,ttab,utab=',itab(i),jtab(i),ttab(i),utab(i)
                   write(iulog,*) 'k=', k, ' pext=', pext(i), ' specext=', specpext(i)
                   write(iulog,*) 'wetvol=', wetvol(i), ' dryvol=', dryvol(i), ' watervol=', watervol(i)
-                  ! write(iulog,*) 'cext=',(cext(i,l),l=1,ncoef)
+                  ! write(iulog,*) 'cext=',(cext(l,i),l=1,ncoef)
```
[the remark this answers is not in the document]
Comment (fvitt) [reply src/physics/cam/modal_aer_opt.F90]: removed

**Review by nusbaume — APPROVED** [review]
Looks great now, thanks!

**Pull request merged**